### DISCUSSION TITLE
Issue when attempting to download binaries to local machine

### DISCUSSION THREAD

**Discussion by Rocco-hash** [telekom-security/tpotce#1323] [Q&A] [answered]
T-pot is so cool thank you and everything is working except this one issue in trying to get the malicious binaries to my RE machine for analysis I receive The following error message  I have changed the permissions of the directory for the logged in user to chmod +rwx but still getting this error if there is a better way to get those binaries out please let me know need some help here thank you all in advance i have looked at discussion issues and havent seen this answered specifically

scp: remote open "/data/dionaea/binaries/0f31231bdfad5283ddf6e": Permission denied

**Comment by Ganesh-403** [comment, marked as answer]
@Rocco-hash 
The error:

```text
scp: remote open "/data/dionaea/binaries/0f31231bdfad5283ddf6e": Permission denied
```

suggests this is a **Linux file permission issue**, rather than a T-Pot or Dionaea issue.

Changing the permissions on the **directory** alone (`chmod +rwx`) is often **not enough**. The file itself must also be readable by the user you're connecting with, and every parent directory in the path must have the execute (`x`) permission so it can be traversed. :contentReference[oaicite:0]{index=0}

I'd recommend checking:

```bash
ls -ld /data
ls -ld /data/dionaea
ls -ld /data/dionaea/binaries
ls -l /data/dionaea/binaries/0f31231bdfad5283ddf6e
```

Pay particular attention to:

- the file owner/group,
- whether your SSH user has read access to the file,
- and whether it can traverse each directory in the path.

If the file is owned by another user (or by the container), a common approach is to copy it to your home directory first and then download it:

```bash
sudo cp /data/dionaea/binaries/<hash> ~/
sudo chown $USER:$USER ~/$(basename <hash>)
```

Then:

```bash
scp -P 64295 <user>@<tpot-host>:~/0f31231bdfad5283ddf6e .
```

This avoids changing permissions on the original malware repository.

Alternatively, if you simply need a one-off copy, you can archive it first:

```bash
sudo tar -czf ~/sample.tar.gz /data/dionaea/binaries/<hash>
```

and then download:

```bash
scp -P 64295 <user>@<tpot-host>:~/sample.tar.gz .
```

One thing to keep in mind is that malware samples are intentionally stored with restrictive permissions in many honeypot deployments to reduce the risk of accidental execution or unauthorized access. I couldn't find any T-Pot documentation recommending making the entire `binaries` directory world-readable, so I'd avoid using `chmod 777` on that directory unless it's a disposable lab environment.

One question that would help identify the exact cause:

- What is the output of:

```bash
ls -l /data/dionaea/binaries/0f31231bdfad5283ddf6e
id
```

That will tell us whether this is simply a file ownership issue or whether the SSH user lacks permission to traverse one of the parent directories.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.

**Comment by Rocco-hash** [comment]
Thank you for your help!!

**Reply by Ganesh-403** [reply]
welcome! :)
